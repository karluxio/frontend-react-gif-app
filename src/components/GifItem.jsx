import PropTypes from "prop-types"

export const GifItem = ({ id, title, url }) => {
  return (

    <div className="card">
      <img src={url} />
      <p>{title}</p>
    </div>

  )
}

GifItem.prototypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
