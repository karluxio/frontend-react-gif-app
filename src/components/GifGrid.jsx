import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {

  const { data, loading, error } = useFetchGifs(category);

  console.log(data);

  return (
    <>
      <h3>{category}</h3>

      {loading && <h3>Loading...</h3>}

      {error && <h3>Error...</h3>}

      <div className="card-grid">
        {!loading && !error && data.map(({ id, title, images }) => (
          <GifItem
            key={id}
            id={id}
            title={title}
            url={images.downsized_medium.url}
          />
        ))}
      </div>

    </>
  )
}
