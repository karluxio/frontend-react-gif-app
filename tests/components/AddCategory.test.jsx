import { render, screen, fireEvent } from "@testing-library/react"
import { AddCategory } from "../../src/components/AddCategory"

describe('testing on <AddCategory />', () => {
  const onNewCategory = jest.fn()

  test('should match against snapshot', () => {
    const { container } = render(<AddCategory onNewCategory={onNewCategory} />)

    expect(container).toMatchSnapshot()
  })

  test('should change input value', () => {
    render(<AddCategory onNewCategory={onNewCategory} />)
    const input = screen.getByRole("textbox")

    fireEvent.input(input, { target: { value: "Goku" } })

    expect(input.value).toBe("Goku")
  })

  test('should call onNewCategory if input has a value', () => {
    const inputValue = "Goku"

    render(<AddCategory onNewCategory={onNewCategory} />)
    const input = screen.getByRole("textbox")
    const form = screen.getByRole("form")

    fireEvent.input(input, { target: { value: inputValue } })
    fireEvent.submit(form)
    expect(input.value).toBe('')
    expect(onNewCategory).toHaveBeenCalled()
    expect(onNewCategory).toHaveBeenCalledTimes(1)
    expect(onNewCategory).toHaveBeenCalledWith(inputValue)
  })

  test('should not call onNewCategory if input is empty', () => {
    const onNewCategory = jest.fn()

    render(<AddCategory onNewCategory={onNewCategory} />)
    const form = screen.getByRole("form")

    fireEvent.submit(form)
    expect(onNewCategory).toHaveBeenCalledTimes(0)
  })


})