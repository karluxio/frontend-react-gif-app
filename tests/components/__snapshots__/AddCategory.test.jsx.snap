// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing on <AddCategory /> should match against snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Add a category"
      type="text"
      value=""
    />
  </form>
</div>
`;
